pkg/secrethub/client.go

```diff
@@ -46,6 +46,23 @@ func NewClient(decrypter Decrypter, authenticator auth.Authenticator, opts *Clie
 	}
 }
 
+// NewClientAWS creates a new SecretHub client that uses AWS STS and KMS to access SecretHub.
+func NewClientAWS(opts *ClientOptions) (Client, error) {
+	decrypter, err := aws.NewKMSDecrypter()
+	if err != nil {
+		return nil, err
+	}
+	client := &clientAdapter{
+		client: newClient(decrypter, auth.NopAuthenticator{}, opts),
```

Comment:
I think we should look into decorating the client with authentication instead of injecting authentication into the client. That's causing us to need a `NopAuthenticator`.